simulator_control/simulator_util.py

```diff
@@ -158,7 +158,7 @@ def Delete(self):
                      preexec_fn=os.setpgrp)
     # The delete command won't delete the simulator log directory.
     if os.path.exists(self.simulator_log_root_dir):
-      shutil.rmtree(self.simulator_log_root_dir)
+      shutil.rmtree(self.simulator_log_root_dir, ignore_errors=True)
```

Comment:
Does ignore_errors force the directory to be deleted, even if there are errors? Or does this flag simply allow execution to continue, even though the directory may not have been deleted?

Reply:
Good catch. I think it only ignores the error. This function accepts a onError handler. Do you think if we need to handle it in that way? Or maybe leaving an empty directory is acceptable.

Reply:
I changed the description to reflect it.

Reply:
I believe we'll need some sort of equivalent to a `-f` solution.

Reply:
Might be worth testing a small python script locally to see if ignore_errors works like `-f` or not.

Reply:
Good suggestion. I tried to reproduce it locally with a python script and got more data. This issue is not easy to reproduce. The documentation of `shutil.rmtree` doesn't say that when it would throw "Directory not empty" error. It could be likely caused by race condition mentioned in https://github.com/eth-cscs/reframe/issues/291. 

I found this issue when a Booting simulator solution was used originally for https://github.com/material-foundation/xctestrunner/pull/2, which caused the deletion process to be slow and `rmtree` can't be finished in time. It doesn't seem to happen for the time being.

I am going to close this PR because `ignore_errors` is not appropriate to be used under this situation.